packages/code/src/main.rs

```diff
@@ -94,13 +94,113 @@ fn transform_code(to: &str) {
+        _ => &ts.themes["InspiredGitHub"],
+    };
+
+    if let Some(syntax) = ss.find_syntax_by_token(lang) {
+        let mut h = HighlightLines::new(syntax, theme);
+        let mut result = String::new();
+        let background_color = theme.settings.background.unwrap();
+        let r = background_color.r;
+        let g = background_color.g;
+        let b = background_color.b;
+        result.push('[');
+        write!(
+            result,
+            r#"{{"name": "raw", "data": "\\definecolor{{background}}{{RGB}}{{{r},{g},{b}}}\n"}},"#
+        )
+        .unwrap();
+        write!(
+            result,
+            r#"{{"name": "raw", "data": "\\begin{{center}}\n\\texttt{{\n\\begin{{codebox}}[colback=background]\n"}},"#
+        )
+        .unwrap();
+        for line in code.split('\n').map(|s| s.trim_end_matches('\r')) {
+            let regions = h.highlight_line(line, &ss).unwrap();
+            let (colors, words): (Vec<_>, Vec<_>) =
+                regions.into_iter().map(|(a, b)| (a.foreground, b)).unzip();
+            for i in 0..words.len() {
+                let r = colors[i].r;
+                let g = colors[i].g;
+                let b = colors[i].b;
+                let word = words[i];
+                let escaped = escape_latex_text(word.to_string(), tab_size);
+                if escaped == " " {
+                    write!(
+                        result,
+                        "{}",
+                        json!({
+                            "name": "raw",
+                            "data": " "
+                        })
+                    )
+                    .unwrap();
+                } else {
+                    write!(result, "{}", json!({
+                        "name": "raw",
+                        "data": format!(r"\textcolor[RGB]{{{r},{g},{b}}}{{{escaped}}}")
+                    })).unwrap();
+                }
+                result.push(',');
+            }
+            write!(result, r#"{{"name": "raw", "data": "\n\\\\"}},"#).unwrap();
+        }
+        write!(
+            result,
+            r#"{{"name": "raw", "data": "\\end{{codebox}}\n}}\n\\end{{center}}\n"}}"#
+        )
+        .unwrap();
+        result.push(']');
+        result
+    } else {
+        eprintln!("Invalid language: {lang}");
+        code.to_string()
+    }
+}
+
+fn escape_latex_text(text: String, tab_size: u64) -> String {
+    let s = text
+        .split('\\')
+        .map(|t| t.replace('{', r"\{").replace('}', r"\}"))
+        .collect::<Vec<String>>()
+        .join(r"\textbacklash{}")
+        .replace('#', r"\#")
+        .replace('$', r"\$")
+        .replace('%', r"\%")
+        .replace('&', r"\&")
+        .replace('_', r"\_")
+        .replace('<', r"\textless{}")
+        .replace('>', r"\textgreater{}")
+        .replace('~', r"\textasciitilde{}")
+        .replace('^', r"\textasciicircum{}")
+        .replace("    ", &format!(r"\hspace*{{{tab_size}mm}}"));
```

Comment:
I think you misinterpreted how `tab_size` should be used. To my understanding, if you find a tab character (`\t`), you should replace it with `tab_size` amount of spaces (in width). I may have misundestood this, but it seems like you are replacing any occurance of four spaces `    ` with `tab_size` millimeters of horizontal space.